plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcConnectionFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.jdbc;
+
+import org.jdbi.v3.core.ConnectionFactory;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class IcebergJdbcConnectionFactory
+        implements ConnectionFactory
+{
+    private final String connectionUrl;
+
+    public IcebergJdbcConnectionFactory(String connectionUrl)
+    {
+        this.connectionUrl = requireNonNull(connectionUrl, "connectionUrl is null");
+    }
+
+    @Override
+    public Connection openConnection()
+            throws SQLException
+    {
+        Connection connection = DriverManager.getConnection(connectionUrl);
```

Comment:
Should we have some connection pool here to draw from?

Reply:
I would skip supporting connection pools in this PR. We can add it later when needed.